.github/workflows/main.yml

```diff
@@ -68,3 +68,18 @@ jobs:
         uses: codecov/codecov-action@v3
         with:
           files: coverage.out
+  test-race:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.x
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Go race test
+        run: go test -race -shuffle=on -timeout 28m ./...
```

Comment:
Why not add the `race` and `shuffle` params to the previous test?

Reply:
@JekaMas already covers it in his workflow in PR #28

Reply:
But aren't we duplicating tests in CI?

Reply:
Ah, you meant just to have a single test workflow with the `race` and `shuffle` flags

Do you see any benefit in keeping these test runs separate? I'm fine with having a single CI run with `race` and other flags enabled

Reply:
The only thing I'm unsure about is if the coverage file will get affected in any way, I'm assuming it won't

cc @ZeljkoBenovic @epikichi

Reply:
@JekaMas maybe is it desirable that they behave differently?

Reply:
'race' flag enables quite a few boundary checks inside golang, eg https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/sync/mutex.go;l=84 (there are tons of them here and there).  This makes execution 3-10 times slower and changes the order of execution that sometimes gives different results then without the flag. That's why I'd have both kinds of tests in the CI: with and without the flag.

Reply:
Resolving this convo because we can tackle this discussion on @JekaMas's PR #28, that originally introduced these race workflows